### ISSUE TITLE
image sft stage error

### ISSUE THREAD

**Issue by zzksdu** [Ola-Omni/Ola#18]
![Image](https://github.com/user-attachments/assets/935fdc61-0265-4692-80bc-71b8a0b48a9c)
i met 2 error when i reproduced your result on image stage.
1. the first error is that the token lenght is exceed your _model_max_length_ in finetune scripts. in this peroid, i only use _Cambrian7M_withsystemprompt_ data and finetune vision tower and mlp module.
2. i got this error: **cur_images_features = image_features[cur_image_idx]** in Ola/ola/model/ola_arch.py when i only use _Cambrian7M_withsystemprompt_ data and finetune vision tower and mlp module.

my setting is as follows:
DATA="xxxxxx/Cambrian_10M/jsons/Cambrian7M_withsystemprompt.json" 
DATA_ROOT="xxxxxxx/Cambrian_10M/"

CHECKPOINT="xxxxx/Qwen2___5-7B-Instruct"
CKPT_MLP='xxxxx/self_pretrain/mm_projector.bin'

VISION_TOWER="xxxxx/google__siglip-so400m-patch14-384"
torchrun \
    --nnodes=1 \
    --node_rank=0 \
    --master_addr=127.0.0.2 \
    --nproc_per_node=${GPUS} \
    --master_port=${MASTER_PORT} \
   ola/train/train.py \
    --deepspeed ./scripts/zero2.json \
    --run_name $EXP_NAME \
    --model_name_or_path $CHECKPOINT \
    --pretrain_mm_mlp_adapter $CKPT_MLP \  #（i got this model from pretrain stage）
    --vision_tower $VISION_TOWER \
    --tune_mm_mlp_adapter False \
    --mm_projector_type ola_mlp \
    --speech_projector_type "linear" \
    --mm_vision_select_layer -1 \
    --mm_use_im_patch_token False \
    --tune_speech_adapter False \
    --tune_mm_vision_resampler True \
    --version qwen_1_5 \
    --data_path $DATA  \
    --data_root_dir $DATA_ROOT \
    --bf16 True \
    --output_dir xxxxx/$EXP_NAME \
    --sample_independently True \
    --fix_speech_encoder True \
    --unfreeze_mm_vision_tower True \
    --speech_encoder "xxxxxx/Ola_speech_encoders/large-v3.pt" \
    --music_encoder "xxxxxxx/Ola_speech_encoders/BEATs_iter3_plus_AS2M_finetuned_on_AS2M_cpt2.pt" \
    --speech_encoder_type "dual" \
    --speech_encoder_hidden_size 2048 \
    --speech_encoder_ds_rate 10 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --gradient_accumulation_steps 1  \
    --evaluation_strategy "no" \
    --save_strategy "steps" \
    --save_steps 1000 \
    --save_total_limit 1 \
    --learning_rate 1e-4 \
    --weight_decay 0.0 \
    --warmup_ratio 0.05 \
    --min_lr_ratio 0.01 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --tf32 True \
    --model_max_length 16384 \
    --gradient_checkpointing True \
    --dataloader_num_workers 8 \
    --frames_upbound 64 \
    --lazy_preprocess True \
    --report_to none

can you give some advices to solve this issues? thanks.

**Comment by liuzuyan** [comment]
1. Please check whether extremely long sentences exist in your training data. In our experiments, occasional notices about overlength may not influence the overall training. 16k max model length is sufficient in most of the circumstances.
2. Please carefully check whether the number of image mark (<image>) matches the total number of given images. If the number of marks is greater than the number of images, the list may be out of range.

**Comment by zzksdu** [comment]
thanks for your advices.
my reproduced loss is not convergence when i reproduced your stage1 result。
my setting is as follows:
EXP_NAME="image_sft_3m_on_buuny_data_0306"
DATA="xxxbunny_v1_1/finetune/bunny_695k.json" 
DATA_ROOT="xxx/bunny_v1_1/finetune/images/"

CHECKPOINT="xxxx/Qwen2___5-7B-Instruct"
CKPT_MLP='xxx/ola_7b_stage1_1_bunny_webui_0303/mm_projector.bin'

VISION_TOWER=xxx/google__siglip-so400m-patch14-384"
echo "start to training for [training]"
torchrun \
    --nnodes=1 \
    --node_rank=0 \
    --master_addr=127.0.0.2 \
    --nproc_per_node=${GPUS} \
    --master_port=${MASTER_PORT} \
   ola/train/train.py \
    --deepspeed ./scripts/zero2.json \
    --run_name $EXP_NAME \
    --model_name_or_path $CHECKPOINT \
    --pretrain_mm_mlp_adapter $CKPT_MLP \
    --vision_tower $VISION_TOWER \
    --tune_mm_mlp_adapter True \
    --tune_mm_vision_resampler True \
    --mm_projector_type ola_mlp \
    --speech_projector_type "linear" \
    --mm_vision_select_layer -1 \
    --mm_use_im_patch_token False \
    --tune_speech_adapter False \
    --version qwen_1_5 \
    --data_path $DATA  \
    --data_root_dir $DATA_ROOT \
    --bf16 True \
    --output_dir xxx/agi/$EXP_NAME \
    --sample_independently True \
    --fix_speech_encoder True \
    --unfreeze_mm_vision_tower True \
    --speech_encoder "xxx/Ola_speech_encoders/large-v3.pt" \
    --music_encoder "xxxx/BEATs_iter3_plus_AS2M_finetuned_on_AS2M_cpt2.pt" \
    --speech_encoder_type "dual" \
    --speech_encoder_hidden_size 2048 \
    --speech_encoder_ds_rate 10 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 4 \
    --per_device_eval_batch_size 1 \
    --gradient_accumulation_steps 1  \
    --evaluation_strategy "no" \
    --save_strategy "steps" \
    --save_steps 1000 \
    --save_total_limit 1 \
    --learning_rate 4e-5 \
    --weight_decay 0.0 \
    --warmup_ratio 0.05 \
    --min_lr_ratio 0.01 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --tf32 True \
    --model_max_length 16384 \
    --gradient_checkpointing True \
    --dataloader_num_workers 8 \
    --frames_upbound 64 \
    --lazy_preprocess True \
    --report_to none
and, i get loss report as:

![Image](https://github.com/user-attachments/assets/94d6d4e7-5231-4c5f-9def-4bc8c157dca8)
could you give me some advices to solve this issue? thanks again @liuzuyan

**Comment by zzksdu** [comment]
@liuzuyan  Another question is where is resampler? I only found a IdentityMap in builder.py of multimodal_resampler folder.

**Comment by liuzuyan** [comment]
> [@eltociear](https://github.com/eltociear) [@dongyh20](https://github.com/dongyh20) [@CharlesCNorton](https://github.com/CharlesCNorton) [@liuzuyan](https://github.com/liuzuyan) Another question is where is resampler? I only found a IdentityMap in builder.py of multimodal_resampler folder.

Hi, we do not use resampler in our model; therefore, we keep it as IdentityMap.
